core/deployment/src/main/java/io/quarkus/deployment/steps/ClassPathSystemPropBuildStep.java

```diff
@@ -46,8 +45,8 @@ public void set(List<SetClassPathSystemPropBuildItem> setCPItems,
 
             }
         }
-        String classPathValue = Stream.concat(parentFirst.stream(), regular.stream()).map(p -> p.toAbsolutePath().toString())
-                .collect(Collectors.joining(":"));
-        recorder.set(classPathValue);
+        List<String> allJarPaths = Stream.concat(parentFirst.stream(), regular.stream()).map(p -> p.toAbsolutePath().toString())
+                .toList();
+        recorder.set(allJarPaths);
```

Comment:
I agree we could merge this as it's better to not fail but I'm thinking that we are definitely not registering the appropriate paths.
From what I can see, it points to the jars in `~/.m2`, which is definitely not expected.

@aloubyansky I suppose we have a way to point to the jars relative to the application? I'm not entirely sure though if the path would be the same in the case of a legacy-jar vs fast-jar vs uber-jar?

Reply:
> I agree we could merge this as it's better to not fail but I'm thinking that we are definitely not registering the appropriate paths

Yup, I mentioned that in the issue as well